.vscode/tasks.json

```
@@ -1,58 +1,43 @@
+
 {
   "tasks": [
     {
+      "label": "WaitForDebugpy",
       "type": "shell",
-      "label": "conda: activate environment",
-      "command": "activate freecad",
+      "command": "python ${workspaceFolder}/.vscode/WaitForPort.py",
+      "group": "none",
       "problemMatcher": [],
-      "detail": "Activate conda environment",
-      "group": "build"
+      "presentation": {
+        "reveal": "never", //silently fail and don't launch the debugger
+        "panel": "dedicated",
+        "close": true,
+        "revealProblems": "never"
+      },
+      "hide": true
     },
     {
-      "type": "cmake",
+      "type": "shell",
       "label": "CMake: configure",
-      "command": "configure",
+      "windows": { "command": "if ('${command:cmake.activeConfigurePresetName}'.ToLower().Contains('conda')) { activate freecad };             cmake ${workspaceFolder} --preset ${command:cmake.activeConfigurePresetName}" },
```

so I'm experimenting on windows and got the conda environment to get properly activated like this:
```suggestion
      "windows": { "command": "${workspaceFolder}/.vscode/scripts/cmake-configure.bat ${workspaceFolder} '${command:cmake.activeConfigurePresetName}'" },
```

then `cmake-configure.bat` has this content:
```cmd
echo %2 | find /i "conda" >nul && (
    call .conda\freecad\condabin\activate.bat
)

cmake %1 --preset %2
```
I went with this solution because powershell doesn't seem to play as nice with conda as cmd does.

Still haven't managed to get it to successfully build though

---

I had noticed the powershell vs cmd issue as well, that's why I've left a commented line with the equivalent command for cmd. However your solution seems much cleaner to me.

What issues still remain to build on windows?